src/web/components/Donut.tsx

```diff
@@ -0,0 +1,189 @@
+import React from 'react';
+import { css } from 'emotion';
+
+import { headline } from '@guardian/src-foundations/typography';
+import { text } from '@guardian/src-foundations/palette';
+
+type Props = {
+    sections: Section[];
+    percentCutout?: number;
+    width?: number;
+    height?: number;
+};
+
+type Section = {
+    label: string;
+    value: number;
+    color: string;
+};
+
+const unitStyles = css`
+    ${headline.medium({ fontWeight: 'bold' })}
+    text-anchor: middle;
+`;
+
+const valueStyles = css`
+    ${headline.small({ fontWeight: 'bold' })}
+    fill: ${text.ctaPrimary};
+    text-anchor: middle;
+`;
+
+const labelStyles = css`
+    ${headline.xxsmall()}
+    fill: ${text.ctaPrimary};
+    text-anchor: middle;
+`;
+
+const withoutZeroSections = (sections: Section[]) =>
+    sections.filter(section => section.value !== 0);
+
+export const Donut = ({
+    sections,
+    percentCutout = 35,
+    width = 300,
+    height = 300,
+}: Props) => {
+    if (withoutZeroSections(sections).length === 1) {
+        console.warn('The Donut component requires at least 2 sections');
+        // TODO: Support showing 100% for a single section
+        return null;
+    }
+
+    if (sections.find(section => section.value === 0)) {
+        console.warn(
+            'Each section in the Donut component must have a non zero value',
+        );
+        // We handle these at the moment by filtering them out using withoutZeroSections()
+        // TODO: Support displaying 0% for sections where value is zero
+    }
+
+    const radius = Math.min(height / 2, width / 2);
+    const cutoutRadius = radius * (percentCutout / 100);
+
+    const totalValue = sections
+        .map(section => section.value)
+        .reduce((prev, next) => prev + next);
+
+    const halfPI = Math.PI / 2;
+    const doublePI = Math.PI * 2;
+
+    const center = {
+        x: width / 2,
+        y: height / 2,
+    };
+
+    // Segments
+    const segments: any[] = [];
+    let segmentAngle;
+    let endRadius;
+    let arc;
+    let outer;
+    let inner;
+    let r;
+    let a;
+    let startRadius = -halfPI;
+
+    withoutZeroSections(sections).forEach(section => {
```

Comment:
i would personally use `map` and defined the `const segments` here rather than `forEach` as `map` expects something to be returned (so you do not need to do `segments.push`)

Reply:
You're right, it's cleaner to use map but when I tried this I ended up returning the object but then needing to still run code inside the loop (updating `v`) after the return statement, which felt wrong

Reply:
if you move `startRadius += (section.value / totalValue) * doublePI;` above `segments.push` and then convert `segments.push` to `return { ... }` the rest should be fine?

Reply:
I think that having the start position reset at the end of the loop is useful. The code does everything using startRadius at one point and _then_ it gets reset at the end. Preserving this order might not be explicitly necessary but in terms of readability I think it helps communicate what's happening.